scripts/test262.js

```diff
@@ -0,0 +1,302 @@
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+import path from "path";
+import fs from "fs";
+import yaml from "js-yaml";
+import os from "os";
+import Integrator from "test262-integrator";
+
+import initializeGlobals from "../lib/globals.js";
+import { AbruptCompletion, ThrowCompletion } from "../lib/completions.js";
+import { DetachArrayBuffer } from "../lib/methods/arraybuffer.js";
+import construct_realm from "../lib/construct_realm.js";
+import { ObjectValue, StringValue } from "../lib/values/index.js";
+import { Realm, ExecutionContext } from "../lib/realm.js";
+import { ToStringPartial } from "../lib/methods/to.js";
+import { Get } from "../lib/methods/get.js";
+
+const testDir = path.join(__dirname, "..", "test", "test262");
+const filters = yaml.safeLoad(fs.readFileSync(path.join(__dirname, "./test262-filters.yml"), "utf8"));
+const EOL = os.EOL;
+
+class TestAttrs {
+  description: string;
+  esid: ?string;
+  es5id: ?string;
+  es6id: ?string;
+  features: ?(string[]);
+  includes: ?(string[]);
+  info: ?string;
+  flags: TestFlags;
+  negative: TestAttrsNegative;
+  skip: boolean;
+}
+
+class TestAttrsNegative {
+  phase: string;
+  type: string;
+}
+
+class TestFlags {
+  generated: boolean;
+  onlyStrict: boolean;
+  noStrict: boolean;
+  async: boolean;
+}
+
+class TestObject {
+  file: string;
+  contents: string;
+  copyright: string;
+  scenario: string; // "default", "strict mode", ...
+  attrs: TestAttrs;
+}
+
+class Result {
+  pass: boolean;
+  error: ?Error;
+  skip: ?boolean;
+  constructor(pass, error, skip): void {
+    this.pass = pass;
+    this.error = error;
+    this.skip = skip;
+  }
+}
+
+class TestResult extends TestObject {
+  result: Result;
+}
+
+function execute(test: TestObject): Result {
+  let { contents, attrs, file, scenario } = test;
+  let { realm } = createRealm(10);
+
+  let strict = scenario === "strict mode";
+
+  // Run the test.
+  try {
+    try {
+      let completion = realm.$GlobalEnv.execute(contents, file);
+      if (completion instanceof ThrowCompletion) throw completion;
+      if (completion instanceof AbruptCompletion) {
+        return new Result(false, new Error("Unexpected abrupt completion"));
+      }
+    } catch (err) {
+      if (err.message === "Timed out") return new Result(false, err);
+      if (!attrs.negative) {
+        throw err;
+      }
+    }
+
+    if (attrs.negative && attrs.negative.type) {
+      throw new Error("Was supposed to error with type " + attrs.negative.type + " but passed");
+    }
+
+    // succeeded
+    return new Result(true);
+  } catch (err) {
+    if (err.value && err.value.$Prototype && err.value.$Prototype.intrinsicName === "SyntaxError.prototype") {
+      // Skip test
```

Comment:
I'm not clear on the intent of this check. Perhaps expand the comment with the rationale?

Reply:
This is not much clear for me as well. I reused most of the logic from the original text execution.

https://github.com/facebook/prepack/blob/master/scripts/test262-runner.js#L994-L995

My plan is to first tackle the stream events but I was taking so long to do a better refactoring for the logics to understand the execution result. This is for sure a place for improvement, but as far as I can tell, this is some sort of test skip for early SyntaxErrors, does it sound right?

Reply:
We do want to skip tests that check for early SyntaxErrors, because dealing with those is out of scope for Prepack. Ideally, there would be a tag in the tests that we can use to exclude them. Absent such a tag, the old framework resorts to running the test and then ignoring the result if it fails with a SyntaxError. It seems from the output of the test run that many tests are still visibly failing because of late SyntaxError detection, so something needs to get fixed here.

Reply:
sure! We have the negative tags in the frontmatter metadata, and I'm using that to capture the filtering.

With the filter on I can work to improve this verification process for the execution results.